src/main/java/io/IO.java

```diff
@@ -0,0 +1,172 @@
+    public static final String COMMAND_TASK_DEADLINE_DELIMITER_REGEX =
+            "\\s+\\" + COMMAND_TASK_DEADLINE_DELIMITER + "\\s+";
+    public static final String COMMAND_TASK_EVENT = "event";
+    public static final String COMMAND_TASK_EVENT_DELIMITER1 = "/from";
+    public static final String COMMAND_TASK_EVENT_DELIMITER2 = "/to";
+    public static final String COMMAND_TASK_EVENT_DELIMITER_REGEX =
+            "\\s+\\" + COMMAND_TASK_EVENT_DELIMITER1 +
+            "\\s+|\\s+\\" + COMMAND_TASK_EVENT_DELIMITER2 + "\\s+";
+
+    /**
+     * Split input into 1st arg and subsequent line.
+     * @param inputLine A line of input direct from user
+     */
+    public static String[] splitCommandAndArgs(String inputLine) {
+        // Trim trailing and beginning whitespace, then split once. Output is [command, args].
+        final String[] split = inputLine.trim().split("\\s+", 2);
+        // Else case: no parameters
+        return split.length == 2 ? split : new String[]{split[0], ""};
+    }
+
+    /**
+     * Input validation for Todo Task. Check if input empty.
+     * @param commandArgs Line input, should simply be the task name
+     * @return the same input, unless blank
+     * @throws DukeException only if blank
+     */
+    public static String processTaskTodo(String commandArgs) throws DukeException {
+        if (commandArgs.isBlank()) {
+            throw new DukeException();
+        }
+        return commandArgs;
+    }
+
+    public static String[] processTaskDeadline(String commandArgs) throws DukeException {
+        String[] deadlineArgs = commandArgs.split(COMMAND_TASK_DEADLINE_DELIMITER_REGEX);
+
+        // Needs to be specifically 2, i.e. [task_name, deadline]
+        // Is this a magic number?
+        if (deadlineArgs.length != 2) {
+            throw new DukeException();
+        }
+        return deadlineArgs;
+    }
+
+    public static String[] processTaskEvent(String commandArgs) throws DukeException {
+        // ensure it contains both.
+        if (!commandArgs.contains(COMMAND_TASK_EVENT_DELIMITER1) ||
+                !commandArgs.contains(COMMAND_TASK_EVENT_DELIMITER2)) {
+            throw new DukeException();
+        }
+
+        // TODO check for inputs like /from abc /from abc /to
+        String[] eventArgs = commandArgs.split(COMMAND_TASK_EVENT_DELIMITER_REGEX);
+
+        // Needs to be specifically 3, i.e. [task_name, from, to]
+        if (eventArgs.length != 3) {
+            throw new DukeException();
+        }
+        return eventArgs;
+    }
+
+    /**
+     * ==============================================================
+     * Below are Outputs for PAPA to print.
+     * Note string format placeholders, so ensure to use printf/format.
+     * ==============================================================
+     */
+    public static final String MESSAGE_HELP = "PAPA is your personal task tracker. Options:\n" +
+            "Add tasks: \n" +
```

Comment:
```suggestion
    public static final String MESSAGE_HELP = "PAPA is your personal task tracker. Options:\n"
            + "Add tasks: \n"
```
Should try to put the operators in front so that readers do not need to keep referring to the previous line